Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/CONNECTSamlAssertionValidator.java

```diff
@@ -301,11 +305,40 @@ protected void checkSignedAssertion(AssertionWrapper assertion, RequestData data
         } catch (WSSecurityException e) {
             if (certs == null && publicKey != null) {
                 LOG.warn("Could not establish trust of the signature's public key because no matching public key "
-                        + "exists in the truststore. Please see GATEWAY-3146 for more details.");
+                    + "exists in the truststore. Please see GATEWAY-3146 for more details.");
             } else {
-                throw e;
+                if (isValidateChainOfTrust() && e.getCause() instanceof CertPathValidatorException){
+                    //handle a special case.  Remove this once all partner upgrade to new chain of trust
+                    CertificateChainValidator chainValidator = CertificateChainValidator.getInstance();
+                    List<X509Certificate> certList = Arrays.asList(certs);
+                    LOG.debug("Detect number of certs in saml {}", certList.size());
+                    for (X509Certificate cert: certList){
+                        String serialNumber = cert.getSerialNumber().toString(16);
```

Comment:
16 is hex value if you want to convert into hexadecimal so that we can display the same cert serial number